### ISSUE TITLE
Enhancement: [prefer-promise-reject-errors] Extend to AbortSignals

### ISSUE THREAD

**Issue by vikingair** [typescript-eslint/typescript-eslint#10717]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/prefer-promise-reject-errors/

### Description

Currently, a user can call `AbortSignal.abort("foo")` or `new AbortController().abort("foo")` with anything. But doing so within a `fetch` request will throw the provided argument. In contrary a `AbortSignal.timeout(1234)` and `new AbortController().abort()` will throw a `DOMException`.

I'd like to ensure that the arguments should be enforced to be instances of `Error` as well.

---

It could make sense to extend rather this rule: https://typescript-eslint.io/rules/only-throw-error

### Fail

```TypeScript
new AbortController().abort("foo");

AbortSignal.abort("foo");
```

### Pass

```TypeScript
new AbortController().abort(new Error("foo"));

AbortSignal.abort(new DOMException("foo"));
```

**Comment by bradzacher** [comment]
I can see value in this as a separate rule -- I don't know if it makes sense to mingle it in with a rule specifically about promises.

I'm personally +0.5 on this -- I think there's value in better standards for this API -- but I also don't think this API is widely-used enough for there to be enough value in us building and owning a rule for it?

cc @typescript-eslint/triage-team

**Comment by JoshuaKGoldberg** [comment]
Yeah I have only ever seen `AbortSignal` used maybe 2-3 times over the last few years. For users of `AbortSignal` this rule does sound useful, but there just aren't that many of them IME. Maybe: `evaluating community engagement`, so if folks are using it, they can let us know?

**Comment by JoshuaKGoldberg** [comment]
It's been over 7 months since this was marked as `evaluating community engagement`. In that time, nobody's expressed any text opinions in favor of this rule. We'll go ahead and close it out as non-actionable for typescript-eslint itself.

If anybody wants to make this rule as a separate plugin, please do! 
* https://typescript-eslint.io/developers/custom-rules: shows how to write custom rules
* https://typescript-eslint.io/developers/eslint-plugins: shows how to package them as plugins

Cheers! 💙
